src/openrct2/actions/WallPlaceAction.cpp

```diff
@@ -431,6 +431,10 @@ GameActions::Result::Ptr WallPlaceAction::Execute() const
     {
         wallElement->SetGhost(true);
     }
+    else
+    {
+        MapInvalidateHeightCache();
```

Comment:
why don't you just invalidate when calling `tile_element_insert`. It doesn't really matter that much about ghosts this isn't the be all end all if it gets invalidated slightly more than required but it would be unfortunate to miss an invalidation.

Reply:
I did it this way because the ghosts blink in and out of existence and cause the height check to restart over and over. It ends up being pretty wasteful and on large parks prevents a height calculation from completing while they flash about.

Reply:
I found I needed to exclude ghosts because they blink in and out of existance quick enough that they prevent the height check from completing on larger parks and it just seemed wasteful. Idealy everything would have been hooked inside of the TileElement, but that didn't work either as I found a lot of code simply edits the clearence height directly via the property rather than the setter method.